automated_packaging/common_functions.pm

```diff
@@ -0,0 +1,165 @@
+        my @prs = @{decode_json($prs_text)};
+
+        foreach my $pr ( @prs ) {
+            my %pr_hash = %$pr;
+
+            if (defined($pr_hash{'merged_at'})) {
+
+                $merged_date = substr($pr_hash{'merged_at'}, 0 ,10);
+
+                if ($merged_date lt $last_date) {
+                    last;
+                }
+
+                $sorted_pr_hash{$pr_hash{'merged_at'}} = $pr_hash{'url'};
+            }
+        }
+
+        $page_number += 1;
+    } while($merged_date gt $last_date);
+
+    my @keys = reverse sort keys(%sorted_pr_hash);
+    my @vals = @sorted_pr_hash{@keys};
+
+    print("PRs has been read in the merge order ..." . "\n");
+
+    return @vals;
+}
+
+# Search for the backport label given a label array
+sub has_backport_label {
+
+    my @labels = @{$_[0]};
+
+    foreach my $label (@labels) {
+        if (ref $label eq ref {}) {
+            %label_hash=%{$label};
+            if ($label_hash{'name'} eq 'backport') {
+                return 1;
+            } 
+        }   
+    }
+
+    return 0;
+}
+
+# Creates changelog entries up to the given last date. It extracts lines starting with DESCRIPTION: from 
+# PR messages to generate these entries. You can use different repo names to generate changelog entries for
+# different repos.
+sub create_release_changelog {
+    my $last_date = @_[0];
+    my $repo_name = @_[1];
+    my $is_point_release = @_[2];
+
+    my @comment_lines = ();
+    my $github_token = get_and_verify_token();
+
+    my @sorted_pr_urls = get_sorted_prs($last_date, $repo_name);
+
+    foreach my $pr_url ( @sorted_pr_urls ) {
+        print("Getting information for " . $pr_url . "\n");
+        my $pr_text = `curl -H "Accept: application/vnd.github.v3.full+json" -X GET --user "$github_token:x-oauth-basic" '$pr_url'`;
+        my %pr_hash = %{decode_json($pr_text)};
+        my $add_to_changelog = 1;
+
+        if ($is_point_release == 1) {
+            my $issue_url = $pr_hash{'issue_url'};
+            my $issue_text = `curl -H "Accept: application/vnd.github.v3.full+json" -X GET --user "$github_token:x-oauth-basic" '$issue_url'`;
+            my %issue_hash = %{decode_json($issue_text)};
+            $add_to_changelog = 0;
+
+            @labels = @{$issue_hash{'labels'}};
+            if (has_backport_label(\@labels) == 1) {
+                $add_to_changelog = 1;
+            }
+        }
+
+        if ($add_to_changelog == 1) {
+            @log_output = split("\n", $pr_hash{'body'});
+            foreach $line (@log_output) {
+                if($line =~ /^DESCRIPTION:.*/) {
+                    $description_part=substr($line ,13 ,-1);
+                    if(length($description_part) > 78) {
+                        `git reset --hard`;
+                        die "Can not add description longer than 78 charachters";
```

Comment:
Let's include which PR has long description to error message, so that user can easily fix it.